### PR TITLE
[*] MO: Possibility to add custom JS and CSS from themeconfigurator

### PR DESCRIPTION
**Pull request by kpodemski**

### CODE REVIEW

**Comment by maximebiloe** [comment]
Hello,

Thanks for your contribution.
There is one little problem : when you want to delete all the custom code, the files are not updated.
Could you correct that and update your PR, please ?

Best regards

**Comment by jeromenadaud** [comment]
Hi @kpodemski 

Will you take in consideration @maximebiloe comment ?

Best regards.

**Comment by kpodemski** [comment]
done

**Review by Progi1984 — APPROVED** [review]

File: themeconfigurator.php [hunk 1]
```diff
@@ -498,6 +500,13 @@ public function getContent()
 			Configuration::updateValue('PS_TC_ACTIVE', (int)Tools::getValue('live_conf'));
 			Configuration::updateValue('PS_GRID_PRODUCT', (int)Tools::getValue('grid_list'));
 			Configuration::updateValue('PS_SET_DISPLAY_SUBCATEGORIES', (int)Tools::getValue('sub_cat'));
+			Configuration::updateValue('PS_TC_CUSTOM_CSS', Tools::getValue('PS_TC_CUSTOM_CSS'), true);
+			Configuration::updateValue('PS_TC_CUSTOM_JS', Tools::getValue('PS_TC_CUSTOM_JS'), true);
+
+			// Custom CSS and JS
+			file_put_contents(_PS_THEME_DIR_.'css/autoload/tc_custom.css', Configuration::get('PS_TC_CUSTOM_CSS'));
```
Comment (matks) [inline themeconfigurator.php:507]: What if `Configuration::get('PS_TC_CUSTOM_CSS')` is empty ? We should not create an empty file, should we ? 🤔

File: themeconfigurator.php [hunk 2]
```diff
@@ -498,6 +500,13 @@ public function getContent()
 			Configuration::updateValue('PS_TC_ACTIVE', (int)Tools::getValue('live_conf'));
 			Configuration::updateValue('PS_GRID_PRODUCT', (int)Tools::getValue('grid_list'));
 			Configuration::updateValue('PS_SET_DISPLAY_SUBCATEGORIES', (int)Tools::getValue('sub_cat'));
+			Configuration::updateValue('PS_TC_CUSTOM_CSS', Tools::getValue('PS_TC_CUSTOM_CSS'), true);
```
Comment (matks) [inline themeconfigurator.php:503]: Is it OK to call `Configuration::updateValue()` even if `PS_TC_CUSTOM_CSS` does not exist ?

I'm not familiar with configuration management 😅 shall we do something like
```
if (Configuration::getValue(PS_TC_CUSTOM_CSS')) {
    Configuration::createValue('PS_TC_CUSTOM_CSS');
}
```

**Review by matks — CHANGES_REQUESTED** [review]
Hi ! A few questions

Also I'm wondering ... this PR enables to add, from the Configuration page of the module, some custom CSS and JSS. But they are not used ? So the theme developer must them include them manually into the theme ?

**Comment by matks** [comment]
@kpodemski Just saw this PR is from 2015. I'm sorry for such a late answer (congratulations to @Progi1984 for digging up all of these PRs).

If you dont want to continue this PR because it is 5 years old, we'll understand 😅

**Comment by kpodemski** [comment]
@matks this module is for PrestaShop 1.6 ;) so... no, there's no reason to continue with it :)

File: themeconfigurator.php (hunk 2 above)
Comment (kpodemski) [reply themeconfigurator.php:503]: @matks yes it's ok, there's no such a thing as "create value", if configuration key doesn't exists it would be created

**Comment by kpodemski** [comment]
@Progi1984 

image

good job with digging ;D

**Comment by matks** [comment]
> @matks this module is for PrestaShop 1.6 ;) so... no, there's no reason to continue with it :)

Goodbye my friend

**Pull request closed without merging**